pallets/loans-ref/src/write_off.rs

```diff
@@ -0,0 +1,180 @@
+use cfg_primitives::Moment;
+use cfg_traits::ops::EnsureFixedPointNumber;
+use codec::{Decode, Encode, MaxEncodedLen};
+use frame_support::{
+	storage::bounded_btree_set::BoundedBTreeSet,
+	traits::tokens::{self},
+	RuntimeDebug,
+};
+use scale_info::TypeInfo;
+use sp_runtime::{traits::Get, ArithmeticError, FixedPointNumber, FixedPointOperand};
+use sp_std::collections::btree_set::BTreeSet;
+use strum::EnumCount;
+
+/// Indicator of when the write off should be applied
+#[derive(
+	Encode,
+	Decode,
+	Clone,
+	PartialEq,
+	Eq,
+	PartialOrd,
+	Ord,
+	TypeInfo,
+	RuntimeDebug,
+	MaxEncodedLen,
+	EnumCount,
+)]
+pub enum WriteOffTrigger {
+	/// Number in days after the maturity date has passed
+	PrincipalOverdueDays(u32),
+
+	/// Seconds since the oracle valuation was last updated
+	OracleValuationOutdated(Moment),
```

Comment:
What do we think we need this for? in which scenario will this be a trigger?

Reply:
When an oracle, has not updated a price for a long time, the chain logic somehow differs from the real value of that price, and affects to the portfolio valuation, so we need to write off that loan in a percentage if it exceeds the maximum updated period for its price.